src/components/cluster/detail/key_pairs.js

```diff
@@ -22,6 +23,12 @@ class ClusterKeyPairs extends React.Component {
     cn_prefix: '',
     cn_prefix_error: null,
     certificate_organizations: '',
+    keypairModal: {
+      visible: false,
```

Comment:
This is what I wanted to avoid somehow.. but I couldn't get it working without code that I felt was just too much hassle. I wanted to fully encapsulate the concern of the modal's state, but maybe that is just not how to do things in React.